superset/superset/db_engine_specs.py

```diff
@@ -1416,6 +1421,74 @@ class SparkSQLEngineSpec(HiveEngineSpec):
 
     engine = 'sparksql'
 
+    @classmethod
+    def _dumps_operation_handle(cls, op_handle):
+        return {
```

Comment:
I didn't check how pyhive types are defined but most probably you can just use `__dict__` attribute instead.

Reply:
It doesn't work because `op_handle.operationId` is an instance of `hive.ttypes.THandleIdentifier`. One way would be to define a custom json encoder/decoder, but IMO it doesn't worth the effort.

Reply:
how about?
```python
dict(op_handle.__dict__, operation_id=op_handle.operationId.__dict__)
```
or
```python
dict(op_handle.__dict__, operation_id={
    'guid': op_handle.operationId.guid.decode('ISO-8859-1'),
    'secret': op_handle.operationId.secret.decode('ISO-8859-1'),
})
```
if decoding is required.

The idea is remove direct dependency on `TOperationHandle` internals.

Reply:
Fixed [here](https://github.com/src-d/sourced-ui/pull/267/commits/3ee0cc30bc89464f7f2b606172d0943deb2bc6e2).

Reply:
looks much nicer!!!